text/0265-event-vision.md

```diff
@@ -0,0 +1,72 @@
+# Event basics
+
+## Motivation
+
+The introduction of Events has been contentious, so we want to write down and agree on a few basics.
+
+### What are events?
+
+Events are a type of log which has a required event name and a specific structure which is implied by that event name.
+
+The event name and its implied structure make events much more suitable for observability compared to traditional logs.
+
+Because of this, the OpenTelemetry project wants to encourage the transition from traditional logs to events.
+
+### OTLP
+
+Since events are a type of log, they share the same OTLP data structure and OTLP pipeline.
+
+### API
+
+OpenTelemetry should have an Event API. This will help to promote the distinction between events and traditional logs,
+and encourage the use of events over traditional logs.
+
+### Interoperability with traditional logs
+
+It should be possible to send events from the Event API to a traditional logging library (e.g. Log4j).
+This allows users to integrate events from the Event API into an existing (non-OpenTelemetry) log stream.
+
+Note: If a user chooses to send events from the Event API to a traditional logging library, and they have
+also chosen to send the logs from their traditional logging library to the OpenTelemetry Logging SDK, then they should
+avoid sending events from the Event API directly to the OpenTelemetry Logging SDK since that would lead to duplicate
+capture of events that were sent from the Event API.
+
+It should be possible to bypass the Event API entirely and emit Events directly via an existing language-specific logging libraries.
+This helps reinforce the idea that events are just a specific type of log.
+
+Note: Because of this, generic event processors should be implemented as Log SDK processors.
+
+OpenTelemetry will recommend that
+[instrumentation libraries](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library)
+use the Event API over emitting events via a traditional logging library in order to give users a simple and consistent
+onboarding story that doesn't involve mixing the two approaches.
+
+OpenTelemetry will recommend that application developers also use the Event API over emitting events via a traditional
+logging library since it avoid accidentally emitting (non-event) traditional logs.
+
+Also, recommending the Event API over emitting events via a traditional logging library makes for a clearer overall
+OpenTelemetry API story - with first class user facing APIs for traces, metrics, and events,
+all suitable for using directly in native instrumentation.
+
+### SDK
+
+The Event SDK needs to support two destinations for events:
+
+* OpenTelemetry Logging SDK
+* Language-specific logging libraries
+
+## Trade-offs and mitigations
+
+TODO
+
+## Prior art and alternatives
+
+TODO
+
+## Open questions
+
+TODO
```

Comment:
I am curious about the direction/intent for opting in/out of specific events. My organization creates user-defined business-events. Today, I use a logging api with the convention of setting `event.name`. This can get clunky when a logger is shared between logs and events. Maybe I want logs at a severity of error but events at a severity of info. Do I create 2nd logger or something else.

This concern might be answered somewhat by https://github.com/open-telemetry/oteps/commit/e9fce2245c1482225684a7cb33579460b6ecf5e0. If we have a dedicated event api then this problem should be solved.

But then I am curious about how on subscribes to specific events (not specific to user-defined events). Do we get everything, register instrumentation scopes (like is done for other signals), something else?. T What relationship does `severity` play here? Is it purely metadata on the record or does it impact what is exported?

Reply:
Having a seperate Event API – which exists today in [the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/event-api.md) – is intended to assist with this. It is then possible to write [event processors](https://github.com/open-telemetry/opentelemetry-specification/pull/4196) which select, modify, and filter events. Otel events include the usual concepts found in logs, such as severity, so it is possible to write processors which utilize those features.

In general, having a pipeline for Otel events that is separate from your existing application logging pipeline is helpful in many situations. This is our justification for providing a complete logging framework for our Otel-specific logging structure, AKA OTel events. Users can choose when and how they integrate Otel events with their existing logs, or optionally leave them completely separate.

Reply:
> I am curious about the direction/intent for opting in/out of specific events

This is definitely important. In addition to @tedsuo's remarks, I think it's also addressed by this (under Future Possibilities):

> The Event API will probably need an `IsEnabled` function based on severity level, scope name, and event name.